File: server/src/main/java/com/server/moviecollection/DatabaseInitializer.java
package com.server.moviecollection;

import com.server.moviecollection.collection.Collection;
import com.server.moviecollection.collection.CollectionRepository;
import com.server.moviecollection.movie.Movie;
import com.server.moviecollection.movie.MovieRepository;
import com.server.moviecollection.series.Series;
import com.server.moviecollection.series.SeriesRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

@Component
public class DatabaseInitializer implements CommandLineRunner {
    @Autowired
    private MovieRepository movieRepository;
    @Autowired
    private SeriesRepository seriesRepository;
    @Autowired
    private CollectionRepository collectionRepository;

    @Override
    public void run(String... args) throws Exception {
        if(movieRepository.count() == 0){
            movieRepository.save(new Movie("Shrek", "/assets/Shrek.jpg", 2001, new String[]{"Andrew Adamson", "Vicky Jenson"}, "1h 30m", false, true));
            movieRepository.save(new Movie("Forrest Gump", "/assets/Forrest_Gump.jpg", 1994, new String[]{"Robert Zemeckis"}, "2h 22m", false, false));
            movieRepository.save(new Movie("Fight Club", "/assets/Fight_Club.jpg", 1999, new String[]{"David Fincher"}, "2h 19m", false, true));
            movieRepository.save(new Movie("The Matrix", "/assets/The_Matrix.jpg", 1999, new String[]{"Lana Wachowski", "Lilly Wachowski"}, "2h 16m", false, false));
            movieRepository.save(new Movie("Inception", "/assets/Inception.jpg", 2010, new String[]{"Christopher Nolan"}, "2h 28m", false, true));
            movieRepository.save(new Movie("Back to the Future", "/assets/Back_To_The_Future.jpg", 1985, new String[]{"Robert Zemeckis"}, "1h 56m", true, true));
            movieRepository.save(new Movie("Back to the Future II", "/assets/Back_To_The_Future_2.jpg", 1989, new String[]{"Robert Zemeckis"}, "1h 48m", true, true));
            movieRepository.save(new Movie("Back to the Future III", "/assets/Back_To_The_Future_3.jpg", 1990, new String[]{"Robert Zemeckis"}, "1h 58m", true, true));
            movieRepository.save(new Movie("Good Will Hunting", "/assets/Good_Will_Hunting.jpg", 1997, new String[]{"Gus Van Sant"}, "2h 06m", true, true));
            movieRepository.save(new Movie("The Lion King", "/assets/The_Lion_King.jpg", 1994, new String[]{"Roger Allers", "Rob Minkoff"}, "1h 28m", false, true));
            movieRepository.save(new Movie("Harry Potter and the Philosopher's Stone", "/assets/Harry_Potter_1.jpg", 2001, new String[]{"Chris Columbus"}, "2h 32m", false, true));
        }

        if(seriesRepository.count() == 0) {
            seriesRepository.save(new Series("Friends", "/assets/Friends.jpg", 1994, 2004, new String[]{"David Crane", "Marta Kauffman"}, "20m", 234, 10, true, true));
            seriesRepository.save(new Series("Rick and Morty", "/assets/Rick_and_Morty.jpg", 2013, 2024, new String[]{"Dan Harmon", "Justin Roiland"}, "23m", 63, 6, false, false));
            seriesRepository.save(new Series("The Office", "/assets/The_Office.jpg", 2005, 2013, new String[]{"Greg Daniels", "Ricky Gervais", "Stephen Merchant"}, "22m", 188, 9, false, true));
            seriesRepository.save(new Series("House M.D.", "/assets/House_MD.jpg", 2004, 2012, new String[]{"David Shore"}, "44m", 176, 8, false, true));
            seriesRepository.save(new Series("Stranger Things", "/assets/Stranger_Things.jpg", 2016, 2024, new String[]{"Matt Duffer", "Ross Duffer"}, "51m", 42, 5, false, true));
            seriesRepository.save(new Series("The Big Bang Theory", "/assets/The_Big_Bang_Theory.jpg", 2007, 2019, new String[]{"Chuck Lorre", "Bill Prady"}, "22m", 280, 12, false, true));
            seriesRepository.save(new Series("Doctor Who", "/assets/Doctor_Who.jpg", 2005, 2023, new String[]{"Sydney Newman"}, "45m", 211, 14, true, false));
            seriesRepository.save(new Series("The Queen's Gambit", "/assets/The_Queen's_Gambit.jpg", 2020, 2020, new String[]{"Scott Frank","Allan Scott"}, "1h", 7, 1, true, true));
            seriesRepository.save(new Series("Battlestar Galactica", "/assets/Battlestar_Galactica.jpg", 2004, 2009, new String[]{"Glen A. Larson","Ronald D. Moore"}, "44m", 74, 4, false, true));
            seriesRepository.save(new Series("Star Trek: Strange New Worlds", "/assets/Star_Trek.jpg", 2022, 2023, new String[]{"Akiva Goldsman","Alex Kurtzman","Jenny Lumet"}, "52m", 30, 3, false, false));
        }

        if(collectionRepository.count() == 0) {
            collectionRepository.save(new Collection("Already Seen"));
            collectionRepository.save(new Collection("Want to watch"));
            collectionRepository.save(new Collection("Favourite"));
            collectionRepository.save(new Collection("Never watch again"));
            collectionRepository.save(new Collection("Untitled"));
            collectionRepository.save(new Collection("Untitled2"));
            collectionRepository.save(new Collection("Untitled3"));
        }
    }
}

File: server_C#/Server_Movie_Collection/Model/DatabaseInitializer.cs
using Microsoft.EntityFrameworkCore;
using Server_Movie_Collection.Entities;
using Server_Movie_Collection.Model.DTO;
using Server_Movie_Collection.Service;

namespace Server_Movie_Collection.Model;

public static class DatabaseInitializer
{
    public static IMovieService MovieService { get; set; }
    public static ISeriesService SeriesService { get; set; }
    public static ICollectionService CollectionService { get; set; }
    public static IDirectorService DirectorService { get; set; }
    
    public static async void SeedData(IServiceProvider serviceProvider)
    {
        await using var context = new MediaCollectionContext(serviceProvider.GetRequiredService<DbContextOptions<MediaCollectionContext>>());
        if (!context.Directors.Any())
        {
            context.Directors.AddRange(
                new Director(){Name = "Andrew Adamson"},
                new Director(){Name = "Vicky Jenson"},
                new Director(){Name = "Robert Zemeckis"},
                new Director(){Name = "David Fincher"},
                new Director(){Name = "Lana Wachowski"},
                new Director(){Name = "Lilly Wachowski"},
                new Director(){Name = "Christopher Nolan"},
                new Director(){Name = "David Crane"},
                new Director(){Name = "Marta Kauffman"},
                new Director(){Name = "Dan Harmon"},
                new Director(){Name = "Justin Roiland"},
                new Director(){Name = "Greg Daniels"},
                new Director(){Name = "Ricky Gervais"},
                new Director(){Name = "Stephen Merchant"},
                new Director(){Name = "David Shore"},
                new Director(){Name = "Matt Duffer"},
                new Director(){Name = "Ross Duffer"},
                new Director(){Name = "Robert Zemeckis"},
                new Director(){Name = "Gus Van Sant"},
                new Director(){Name = "Roger Allers"},
                new Director(){Name = "Rob Minkoff"},
                new Director(){Name = "Chris Columbus"},
                new Director(){Name = "Chuck Lorre"},
                new Director(){Name = "Bill Prady"},
                new Director(){Name = "Sydney Newman"},
                new Director(){Name = "Scott Frank"},
                new Director(){Name = "Allan Scott"},
                new Director(){Name = "Glen A. Larson"},
                new Director(){Name = "Ronald D. Moore"},
                new Director(){Name = "Akiva Goldsman"},
                new Director(){Name = "Alex Kurtzman"},
                new Director(){Name = "Jenny Lumet"}
            );
            await context.SaveChangesAsync();
        }

        if (context.Directors.Any() && !context.Movies.Any())
        {
            context.Movies.AddRange(
                new Movie(){Title = "Shrek", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Andrew Adamson")),context.Directors.First(director => director.Name.Equals("Vicky Jenson"))}, PosterUrl = "/assets/Shrek.jpg", Favourite = false, Watched = true, Length = "1h 30m", Year = 2001},
                new Movie(){Title = "Forrest Gump", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Robert Zemeckis"))}, PosterUrl = "/assets/Forrest_Gump.jpg", Favourite = false, Watched = true, Length = "2h 22m", Year = 1994},
                new Movie(){Title = "Fight Club", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("David Fincher"))}, PosterUrl = "/assets/Fight_Club.jpg", Favourite = false, Watched = true, Length = "2h 19m", Year = 1999},
                new Movie(){Title = "The Matrix", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Lana Wachowski")), context.Directors.First(director => director.Name.Equals("Lilly Wachowski"))}, PosterUrl = "/assets/The_Matrix.jpg", Favourite = false, Watched = true, Length = "2h 16m", Year = 1999},
                new Movie(){Title = "Inception", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Christopher Nolan"))}, PosterUrl = "/assets/Inception.jpg", Favourite = false, Watched = true, Length = "2h 28m", Year = 2010},
                new Movie(){Title = "Back to the Future", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Robert Zemeckis"))}, PosterUrl = "/assets/Back_To_The_Future.jpg", Favourite = false, Watched = true, Length = "1h 56m", Year = 1985},
                new Movie(){Title = "Back to the Future II", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Robert Zemeckis"))}, PosterUrl = "/assets/Back_To_The_Future_2.jpg", Favourite = false, Watched = true, Length = "1h 48m", Year = 1989},
                new Movie(){Title = "Back to the Future III", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Robert Zemeckis"))}, PosterUrl = "/assets/Back_To_The_Future_3.jpg", Favourite = false, Watched = true, Length = "1h 58m", Year = 1990},
                new Movie(){Title = "Good Will Hunting", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Gus Van Sant"))}, PosterUrl = "/assets/Good_Will_Hunting.jpg", Favourite = false, Watched = true, Length = "2h 06m", Year = 1997},
                new Movie(){Title = "The Lion King", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Roger Allers")), context.Directors.First(director => director.Name.Equals("Rob Minkoff"))}, PosterUrl = "/assets/The_Lion_King.jpg", Favourite = false, Watched = true, Length = "1h 28m", Year = 1994},
                new Movie(){Title = "Harry Potter and the Philosopher's Stone", Directors = new List<Director>(){context.Directors.First(director => director.Name.Equals("Chris Columbus"))}, PosterUrl = "/assets/Harry_Potter_1.jpg", Favourite = false, Watched = true, Length = "2h 32m", Year = 2001}
            );

            await context.SaveChangesAsync();
        }

        if (!context.Collections.Any())
        {
            context.Collections.AddRange(
                new MediaCollection(){Title = "Already Seen"},
                new MediaCollection(){Title = "Want to watch"},
                new MediaCollection(){Title = "Favourite"},
                new MediaCollection(){Title = "Never watch again"},
                new MediaCollection(){Title = "Untitled"},
                new MediaCollection(){Title = "Untitled2"},
                new MediaCollection(){Title = "Untitled3"}
            );

            await context.SaveChangesAsync();
        }
    }
}